src/main/java/fi/metatavu/kuntaapi/server/discover/Updater.java

```diff
@@ -0,0 +1,54 @@
+package fi.metatavu.kuntaapi.server.discover;
+
+import java.util.Iterator;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.annotation.Resource;
+import javax.ejb.EJBContext;
+import javax.ejb.Singleton;
+import javax.ejb.Startup;
+import javax.enterprise.concurrent.ManagedScheduledExecutorService;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Any;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+
+import fi.metatavu.kuntaapi.server.tasks.Job;
+
+@Singleton
+@ApplicationScoped
+@Startup
+public class Updater {
+
+  @Inject
+  @Any
+  private Instance<AbstractDiscoverJob> jobs;
+
+  @Resource
+  private ManagedScheduledExecutorService managedScheduledExecutorService;
+  
+  @Resource
+  private EJBContext ejbContext;
+  
+  @PostConstruct
+  public void postConstruct() {
+    Iterator<AbstractDiscoverJob> jobIterator = jobs.iterator();
+    while (jobIterator.hasNext()) {
+      startJob(jobIterator.next());
+    }
+  }
+  
+  @PreDestroy
+  public void preDestroy() {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)